### PR TITLE
4pplet keyboards

### PR DESCRIPTION
**Pull request by 4pplet**
## Description

Adding masterwork classy_tkl keyboard. Information about group buy: https://geekhack.org/index.php?topic=108263.0

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/masterwork/classy_tkl/rev_a/rules.mk [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```
Comment (drashna) [inline keyboards/masterwork/classy_tkl/rev_a/rules.mk:41]:
```suggestion
```

File: keyboards/masterwork/classy_tkl/rev_a/rules.mk [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+BOOTLOADER = atmel-dfu
+
+# If you don't know the bootloader type, then you can specify the
+# Boot Section Size in *bytes* by uncommenting out the OPT_DEFS line
+#   Teensy halfKay      512
+#   Teensy++ halfKay    1024
+#   Atmel DFU loader    4096
+#   LUFA bootloader     4096
+#   USBaspLoader        2048
+# OPT_DEFS += -DBOOTLOADER_SIZE=4096
```
Comment (drashna) [inline keyboards/masterwork/classy_tkl/rev_a/rules.mk:21]:
```suggestion
BOOTLOADER = atmel-dfu

```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by 4pplet** [comment]
Thanks @drashna ! :)

**Comment by 4pplet** [comment]
Fixed spelling error for masterwork to masterworks, sorry about that!

File: keyboards/masterworks/classy_tkl/info.json [hunk 3]
```diff
@@ -0,0 +1,295 @@
+{
+    "keyboard_name": "Classy TKL",
+    "url": "https://geekhack.org/index.php?topic=105933",
+    "maintainer": "qmk",
+    "width": 18.25,
+    "height": 6.5,
+    "layouts": {
+        "LAYOUT_all": {
+            "key_count": 89,
```
Comment (drashna) [inline keyboards/masterworks/classy_tkl/info.json:9]:
```suggestion
```

File: keyboards/masterworks/classy_tkl/info.json [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+                { "label": "Alt", "x": 11, "y": 5.5, "w": 1.5 },
+                { "label": "Win", "x": 12.5, "y": 5.5, "w": 1 },
+                { "label": "Ctrl", "x": 13.5, "y": 5.5, "w": 1.5 },
+                { "label": "Left", "x": 15.25, "y": 5.5 },
+                { "label": "Down", "x": 16.25, "y": 5.5 },
+                { "label": "Right", "x": 17.25, "y": 5.5 }
+            ]
+        },
+        "LAYOUT_tkl_ansi_wkl": {
+            "key_count": 84,
```
Comment (drashna) [inline keyboards/masterworks/classy_tkl/info.json:103]:
```suggestion
```

File: keyboards/masterworks/classy_tkl/info.json [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+                { "label": "Space", "x": 4, "y": 5.5, "w": 7 },
+                { "label": "Alt", "x": 11, "y": 5.5, "w": 1.5 },
+                { "label": "Ctrl", "x": 13.5, "y": 5.5, "w": 1.5 },
+                { "label": "Left", "x": 15.25, "y": 5.5 },
+                { "label": "Down", "x": 16.25, "y": 5.5 },
+                { "label": "Right", "x": 17.25, "y": 5.5 }
+            ]
+        },
+        "LAYOUT_tkl_iso_wkl": {
+            "key_count": 85,
```
Comment (drashna) [inline keyboards/masterworks/classy_tkl/info.json:192]:
```suggestion
```

File: keyboards/masterworks/classy_tkl/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,13 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/masterworks/classy_tkl/keymaps/default/keymap.c:1]: Could you add a license header, here as well?

File: keyboards/masterworks/classy_tkl/keymaps/default_tkl_ansi_wkl/keymap.c [hunk 7]
```diff
@@ -0,0 +1,13 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/masterworks/classy_tkl/keymaps/default_tkl_ansi_wkl/keymap.c:1]: And here?

File: keyboards/masterworks/classy_tkl/keymaps/default_tkl_iso_wkl/keymap.c [hunk 8]
```diff
@@ -0,0 +1,13 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/masterworks/classy_tkl/keymaps/default_tkl_iso_wkl/keymap.c:1]: And here?

File: keyboards/masterworks/classy_tkl/keymaps/via/keymap.c [hunk 9]
```diff
@@ -0,0 +1,34 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/masterworks/classy_tkl/keymaps/via/keymap.c:1]: Also, here?

File: keyboards/masterworks/classy_tkl/info.json [hunk 10]
```diff
@@ -0,0 +1,292 @@
...
+                    "ks": [
+                        [0, 0],
+                        [1.5, 0],
+                        [1.5, 2],
+                        [0.25, 2],
+                        [0.25, 1],
+                        [0, 1],
+                        [0, 0]
+                    ]
+                },
```
Comment (noroadsleft) [inline keyboards/masterworks/classy_tkl/info.json:266]: QMK Configurator's UI code actually doesn't support `ks`, so we do this:

```suggestion
                { "label": "Enter", "x": 13.75, "y": 2.5, "w": 1.25, "h": 2 },
```

... and it renders like an ISO Enter.

File: keyboards/masterworks/classy_tkl/readme.md [hunk 11]
```diff
@@ -0,0 +1,15 @@
+# Classy TKL
+
+![Classy TKL](https://i.imgur.com/p1dxfYK.jpg)
```
Comment (noroadsleft) [inline keyboards/masterworks/classy_tkl/readme.md:3]: Swaps this image for a smaller version.

```suggestion
![Classy TKL](https://i.imgur.com/p1dxfYKl.jpg)
```

File: keyboards/masterworks/classy_tkl/readme.md [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+
+A short description of the keyboard/project
+
+Keyboard Maintainer: [Mathias Andersson](https://github.com/wraul)  
+Hardware Supported: [Classy TKL PCB](https://github.com/4pplet/classyTKL)  
+Hardware Availability: https://geekhack.org/index.php?topic=105933
```
Comment (noroadsleft) [inline keyboards/masterworks/classy_tkl/readme.md:9]:
```suggestion
* Keyboard Maintainer: [Mathias Andersson](https://github.com/wraul)
* Hardware Supported: [Classy TKL PCB](https://github.com/4pplet/classyTKL)
* Hardware Availability: https://geekhack.org/index.php?topic=105933
```

File: keyboards/masterworks/classy_tkl/rev_a/config.h [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID 0x4D57  // "MW" = Masterworks
+#define PRODUCT_ID 0x0001
+#define DEVICE_VER 0x0001
+#define MANUFACTURER Masterworks
+#define PRODUCT Classy TKL
+#define DESCRIPTION A classy keyboard
```
Comment (noroadsleft) [inline keyboards/masterworks/classy_tkl/rev_a/config.h:28]: `DESCRIPTION` isn't used anywhere in QMK anymore.

```suggestion
```

File: keyboards/masterworks/classy_tkl/rev_a/rev_a.c [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+  // put your per-action keyboard code here
+  // runs for every action, just before processing by the firmware
+
+  return process_record_user(keycode, record);
+}
+
+*/
+
```
Comment (noroadsleft) [inline keyboards/masterworks/classy_tkl/rev_a/rev_a.c:52]:
```suggestion
```

File: keyboards/masterworks/classy_tkl/rev_a/rev_a.h [hunk 15]
```diff
@@ -0,0 +1,79 @@
...
+    { k30, k31,   k32,   k33, k34,   k35,   k36,   k37, k38,   k39,   k3A,   k3B, k3C,   k3D,   KC_NO, KC_NO, KC_NO }, \
+    { k40, KC_NO, k42,   k43, k44,   k45,   k46,   k47, k48,   k49,   k4A,   k4B, k4C,   KC_NO, KC_NO, k4F,   KC_NO }, \
+    { k50, KC_NO, KC_NO, k53, KC_NO, KC_NO, KC_NO, k57, KC_NO, KC_NO, KC_NO, k5B, KC_NO, k5D,   k5E,   k5F,   k5G },   \
+}
+
+#define LAYOUT_tkl_iso_wkl( \
+    k00,      k02, k03, k04, k05,      k06, k07, k08, k09,      k0A, k0B, k0C, k0D,      k0E, k0F, k0G, \
+    k10, k11, k12, k13, k14, k15, k16, k17, k18, k19, k1A, k1B, k1C,           k3D,      k1E, k1F, k1G, \
+    k20,      k21, k22, k23, k24, k25, k26, k27, k28, k29, k2A, k2B, k2C,      k2D,      k2E, k2F, k2G, \
+    k30,      k31, k32, k33, k34, k35, k36, k37, k38, k39, k3A, k3B, k3C,                               \
```
Comment (noroadsleft) [inline keyboards/masterworks/classy_tkl/rev_a/rev_a.h:67]: Based on the ISO keymap you've provided, I think this needs to be:

```suggestion
    k20,      k21, k22, k23, k24, k25, k26, k27, k28, k29, k2A, k2B, k2C,                k2E, k2F, k2G, \
    k30,      k31, k32, k33, k34, k35, k36, k37, k38, k39, k3A, k3B, k3C,      k2D,                     \
```

QMK custom is that ISO Enter is to the end of the home row.

File: keyboards/masterworks/classy_tkl/rev_a/rules.mk [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
```
Comment (noroadsleft) [inline keyboards/masterworks/classy_tkl/rev_a/rules.mk:22]:
```suggestion
```

File: keyboards/masterworks/classy_tkl/rev_a/rules.mk [hunk 17]
```diff
@@ -0,0 +1,27 @@
...
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
```
Comment (noroadsleft) [inline keyboards/masterworks/classy_tkl/rev_a/rules.mk:24]:
```suggestion
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Comment by noroadsleft** [comment]
Still two comments unaddressed on my previous review.

**Comment by 4pplet** [comment]
Thanks for the suggestions, good catch on the ISO-enter!

**Review by noroadsleft — APPROVED** [review]

**Comment by 4pplet** [comment]
Any update on this? All seems done? @drashna :)

**Review by drashna — APPROVED** [review]
Thanks!

**Pull request merged**